From 61f91cf0bb30455fc11a6c72e7161d7da34ea01e Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 10 May 2012 13:00:34 +0000
Subject: [PATCH 0201/1000] cardscan.lib; V1.0.0.0 Mantis 0023248: airberlin:
 new topbonus cards for SKK added to cards.cfg warnings fixed version added

git-svn-id: svn://localhost/SelfServiceCommon/trunk@332 90b65887-3827-0410-9a23-83215b262276
---
 .../Services/cpp/CardScanner/cards.cfg        | 114 ++++++++++++++++--
 .../Services/cpp/CardScanner/makefile.mak     |   1 +
 .../cpp/CardScanner/src/cardtools.cpp         | 112 ++++++++---------
 .../cpp/CardScanner/src/local_def.cpp         |  12 ++
 4 files changed, 176 insertions(+), 63 deletions(-)
 create mode 100644 SelfServiceCommon/Services/cpp/CardScanner/src/local_def.cpp

diff --git a/SelfServiceCommon/Services/cpp/CardScanner/cards.cfg b/SelfServiceCommon/Services/cpp/CardScanner/cards.cfg
index b881ef13..b14d425c 100644
--- a/SelfServiceCommon/Services/cpp/CardScanner/cards.cfg
+++ b/SelfServiceCommon/Services/cpp/CardScanner/cards.cfg
@@ -248,7 +248,7 @@
         // [992229847138113=0502102]
         // []
 
-        card24 = "LH-M&M";  \
+        card24 = "LH-M+M";  \
             item: "99200",     2, "",       1; \
             item: "102",       2, "=",      5; \
             num:  15    ,     2, "",       1; \
@@ -260,7 +260,7 @@
         // [992229847138113=0502102]
         // []
 
-        card25 = "LH-M&M-C";  \
+        card25 = "LH-M+M-C";  \
             item: "99220",     2, "",       1; \
             item: "102",       2, "=",      5; \
             num:  15    ,     2, "",       1; \
@@ -392,28 +392,28 @@
             item: "11",       2, "",       27; \
             num:   9    ,     2, "",       29; \
             expd: 4     ,     2, "",       16; \
-            name: 26    ,     1, "^",      1;
+            name: 20    ,     1, "^",      1;
 
         card44 = "SK-EBS";  \
             item: "2117",     2, "",       1; \
             item: "12",       2, "",       27; \
             num:   9    ,     2, "",       29; \
             expd: 4     ,     2, "",       16; \
-            name: 26    ,     1, "^",      1;
+            name: 20    ,     1, "^",      1;
 
         card45 = "SK-EBG";  \
             item: "2117",     2, "",       1; \
             item: "13",       2, "",       27; \
             num:   9    ,     2, "",       29; \
             expd: 4     ,     2, "",       16; \
-            name: 26    ,     1, "^",      1;
+            name: 20    ,     1, "^",      1;
 
         card46 = "SK-EBP";  \
             item: "2117",     2, "",       1; \
             item: "14",       2, "",       27; \
             num:   9    ,     2, "",       29; \
             expd: 4     ,     2, "",       16; \
-            name: 26    ,     1, "^",      1;
+            name: 20    ,     1, "^",      1;
 
         card47 = "NZ-GE";  \
             item: "504618001",1, "",       1; \
@@ -713,7 +713,7 @@
             expd: 4     ,     1, "^^",     1; \
             name: 20    ,     1, "^",      1;
 
-        card93 = "AB-TopBonus";  \
+        card93 = "AB_TopBonus";  \
             item: "AB",       1, "",       1; \
             item: "0000",     2, "=",      5; \
             num:  9     ,     2, "",       1; \
@@ -731,3 +731,103 @@
             item: "1",        2, "",       1; \
             num:  6  ,        2, "",       1; \
             name: 20  ,       1, "^",      1;
+
+        card96 = "tuifly"; \
+            item: "HF"  ,     1, "",       2; \
+            num:  16    ,     1, "",       6; \
+            expd: 4     ,     2, "",       13; \
+            name: 23    ,     1, "",       27;
+
+        card97 = "AB-C";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "C", 1, "^^", 22; \
+           num:  9 ,  1, "^^", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "^",  0;
+
+        card98 = "AB-S";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "S", 1, "^^", 22; \
+           num:  9 ,  1, "^^", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "^",  0;
+
+        card99 = "AB-G";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "G", 1, "^^", 22; \
+           num:  9 ,  1, "^^", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "^",  0;
+
+
+        card100  = "AB-C-NEU";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "C", 1, "##", 22; \
+           num:  9 ,  1, "##", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "#",  0;
+
+        card101  = "AB-S-NEU";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "S", 1, "##", 22; \
+           num:  9 ,  1, "##", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "#",  0;
+
+        card102  = "AB-G-NEU";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "G", 1, "##", 22; \
+           num:  9 ,  1, "##", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "#",  0;
+
+        card103  = "AB-P-NEU";  \
+           item: "4", 1, "",   2; \
+           item: "4", 2, "",   1; \
+           item: "P", 1, "##", 22; \
+           num:  9 ,  1, "##", 13; \
+           expd: 4  , 2, "=",  0; \
+           name: 26 , 1, "#",  0;
+
+        card104 = "AB-TopBonus-C";  \
+           item: "AB",       1, "",       1; \
+           item: "#",        1, "",       13; \
+           item: "#",        1, "",       31; \
+           item: "C",        1, "",       65; \
+           num:  9     ,     1, "",       4; \
+           expd: 4     ,     1, "##",     1; \
+           name: 20    ,     1, "#",      1;
+
+        card105 = "AB-TopBonus-S";  \
+           item: "AB",       1, "",       1; \
+           item: "#",        1, "",       13; \
+           item: "#",        1, "",       31; \
+           item: "S",        1, "",       65; \
+           num:  9     ,     1, "",       4; \
+           expd: 4     ,     1, "##",     1; \
+           name: 20    ,     1, "#",      1;
+
+        card106 = "AB-TopBonus-G";  \
+           item: "AB",       1, "",       1; \
+           item: "#",        1, "",       13; \
+           item: "#",        1, "",       31; \
+           item: "G",        1, "",       65; \
+           num:  9     ,     1, "",       4; \
+           expd: 4     ,     1, "##",     1; \
+           name: 20    ,     1, "#",      1;
+
+        card107 = "AB-TopBonus-P";  \
+           item: "AB",       1, "",       1; \
+           item: "#",        1, "",       13; \
+           item: "#",        1, "",       31; \
+           item: "P",        1, "",       65; \
+           num:  9     ,     1, "",       4; \
+           expd: 4     ,     1, "##",     1; \
+           name: 20    ,     1, "#",      1;
+
diff --git a/SelfServiceCommon/Services/cpp/CardScanner/makefile.mak b/SelfServiceCommon/Services/cpp/CardScanner/makefile.mak
index 8705868e..00fdb479 100644
--- a/SelfServiceCommon/Services/cpp/CardScanner/makefile.mak
+++ b/SelfServiceCommon/Services/cpp/CardScanner/makefile.mak
@@ -15,6 +15,7 @@ MY_CFLAGS = \
 MY_LIB1 = $(LIB_DST_CARDSCAN)
     
 MY_LIB1_OBJS = \
+    $(_OBJ)\local_def.obj \
     $(_OBJ)\cardtools.obj \
     $(_OBJ)\cardscan.obj
 
diff --git a/SelfServiceCommon/Services/cpp/CardScanner/src/cardtools.cpp b/SelfServiceCommon/Services/cpp/CardScanner/src/cardtools.cpp
index 1c223f5b..fc86390f 100644
--- a/SelfServiceCommon/Services/cpp/CardScanner/src/cardtools.cpp
+++ b/SelfServiceCommon/Services/cpp/CardScanner/src/cardtools.cpp
@@ -95,7 +95,7 @@ bool     fFound;
     if(lSepLen && pCard->lExpLen)
     {
       // check relative to separators (behind separators!)
-	  
+
       strcpy(szTmp,pCardData->szTrack[pCard->lExpTrack-1]);
       pStr = szTmp;
 
@@ -105,22 +105,22 @@ bool     fFound;
           pStr++;
       }
     }
-	else if(lSepLen > 1 && !pCard->lExpLen)
-	{
-		// check between separators
-		
-		strcpy(szTmp,pCardData->szTrack[pCard->lExpTrack-1]);
-		pStr = szTmp;
-
-		for(i = 0; i < lSepLen; i++)
-		{
-			if((pStr = strchr(pStr,pCard->szExpSep[i])) != NULL)
-			{
-				pStr++;
-				break;
-			}
-		}
-	}
+    else if(lSepLen > 1 && !pCard->lExpLen)
+    {
+        // check between separators
+
+        strcpy(szTmp,pCardData->szTrack[pCard->lExpTrack-1]);
+        pStr = szTmp;
+
+        for(i = 0; i < lSepLen; i++)
+        {
+            if((pStr = strchr(pStr,pCard->szExpSep[i])) != NULL)
+            {
+                pStr++;
+                break;
+            }
+        }
+    }
     else
     {
       // check absolute position
@@ -215,28 +215,28 @@ bool     fFound;
           pStr++;
       }
     }
-	else if(lSepLen > 1 && !pCard->lNumLen)
-	{
-		// check between separators
-		
-		strcpy(szTmp,pCardData->szTrack[pCard->lNumTrack-1]);
-		pStr = szTmp;
-
-		for(i = 0; i < lSepLen; i++)
-		{
-			if((pStr = strchr(pStr,pCard->szNumSep[i])) != NULL)
-			{
-				pStr++;
-				break;
-			}
-		}
-	}
+    else if(lSepLen > 1 && !pCard->lNumLen)
+    {
+        // check between separators
+
+        strcpy(szTmp,pCardData->szTrack[pCard->lNumTrack-1]);
+        pStr = szTmp;
+
+        for(i = 0; i < lSepLen; i++)
+        {
+            if((pStr = strchr(pStr,pCard->szNumSep[i])) != NULL)
+            {
+                pStr++;
+                break;
+            }
+        }
+    }
     else
     {
       // check absolute position
 
       if(pCard->lNumTrack >= 1)
-      {      
+      {
         strcpy(szTmp,pCardData->szTrack[pCard->lNumTrack-1]);
         pStr = szTmp;
       }
@@ -329,22 +329,22 @@ bool  fFound;
           pStr++;
       }
     }
-	else if(lSepLen > 1 && !pCard->lNameLen)
-	{
-		// check between separators
-		
-		strcpy(szTmp,pCardData->szTrack[pCard->lNameTrack-1]);
-		pStr = szTmp;
-
-		for(i = 0; i < lSepLen; i++)
-		{
-			if((pStr = strchr(pStr,pCard->szNameSep[i])) != NULL)
-			{
-				pStr++;
-				break;
-			}
-		}
-	}
+    else if(lSepLen > 1 && !pCard->lNameLen)
+    {
+        // check between separators
+
+        strcpy(szTmp,pCardData->szTrack[pCard->lNameTrack-1]);
+        pStr = szTmp;
+
+        for(i = 0; i < lSepLen; i++)
+        {
+            if((pStr = strchr(pStr,pCard->szNameSep[i])) != NULL)
+            {
+                pStr++;
+                break;
+            }
+        }
+    }
     else
     {
       // check absolute position
@@ -483,8 +483,8 @@ bool            fMatch;
         if(pStr && *pStr)
         {
           // set offset
-		  
-		  if (strlen(pStr)<=pMatch->lPos)
+
+          if (static_cast<long>(strlen(pStr))<=pMatch->lPos)
           {
             fMatch = false;
             break;
@@ -511,10 +511,10 @@ bool            fMatch;
           }
         }
         else
-		{
+        {
           fMatch = false;
-		  break;
-		}
+          break;
+        }
 
         // check next entry
 
@@ -858,7 +858,7 @@ char szToken[100];
 
      strcpy(szToken,pszToken);
 
-     strlwr(szToken);
+     _strlwr(szToken);
 
      if(strstr(szToken,"item") == szToken)
      {
diff --git a/SelfServiceCommon/Services/cpp/CardScanner/src/local_def.cpp b/SelfServiceCommon/Services/cpp/CardScanner/src/local_def.cpp
new file mode 100644
index 00000000..c644083a
--- /dev/null
+++ b/SelfServiceCommon/Services/cpp/CardScanner/src/local_def.cpp
@@ -0,0 +1,12 @@
+#include "Tools/VERSIONLIB.h"
+
+VERSION_ID_DT(cardscanner, 1, 0, 0, 0);
+
+/*! \page cardscan.lib, cardscan.lib - Release Notes
+ *
+ *  \date 10.05.2012 \li V1.0.0.0
+ *                   \li SteT extension Mantis 0023248: airberlin: new topbonus cards for SKK added to cards.cfg
+ *                            enhancement warnings fixed, version added
+ *
+ */
+
-- 
2.41.0.windows.1

